kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/visitor/linux/WpaSupplicantConfigWriter.java

```diff
@@ -82,18 +85,41 @@ public void visit(NetworkConfiguration config) throws KuraException {
                 .getModifiedNetInterfaceConfigs();
 
         for (NetInterfaceConfig<? extends NetInterfaceAddressConfig> netInterfaceConfig : netInterfaceConfigs) {
-            if (netInterfaceConfig.getType() == NetInterfaceType.WIFI) {
-                // ignore 'mon' interface
-                if (netInterfaceConfig.getName().startsWith("mon.")) {
-                    continue;
-                }
+
+            // skip monitor interface
+            if (netInterfaceConfig.getName().startsWith("mon.")) {
+                continue;
+            }
+
+            Optional<WifiMode> wifiMode = getWifiMode(netInterfaceConfig);
+
+            if (wifiMode.isPresent() && (wifiMode.get() == WifiMode.INFRA || wifiMode.get() == WifiMode.ADHOC)
+                    && netInterfaceConfig.getType() == NetInterfaceType.WIFI) {
 
                 writeConfig(netInterfaceConfig);
             }
         }
 
     }
 
+    private Optional<WifiMode> getWifiMode(NetInterfaceConfig<? extends NetInterfaceAddressConfig> netInterfaceConfig)
+            throws KuraException {
+
+        Optional<WifiMode> wifiMode = Optional.empty();
+
+        Optional<NetInterfaceConfig<? extends NetInterfaceAddressConfig>> wifiConfig = netInterfaceConfig instanceof WifiInterfaceConfigImpl
+                ? Optional.of(netInterfaceConfig)
+                : Optional.empty();
+
+        if (wifiConfig.isPresent()) {
+            WifiInterfaceAddressConfig wifiInterfaceAddressConfig = (WifiInterfaceAddressConfig) ((WifiInterfaceConfigImpl) wifiConfig
+                    .get()).getNetInterfaceAddressConfig();
+            wifiMode = Optional.ofNullable(wifiInterfaceAddressConfig.getMode());
+        }
+
+        return wifiMode;
+    }
+
```

Comment:
Can this method be extracted somewhere to avoid duplication?

Reply:
I thought to do that, but I can't get where to put it, any idea?

Reply:
Maybe in `WifiConfigWriter`? As a static protected method?

Reply:
i've created a common abstract class with the shared methods.